griptape/drivers/image_generation/openai_dalle_image_generation_driver.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+import base64
+import openai
+from typing import Optional, Literal, Union, get_args
+from attr import field, Factory, define
+from griptape.artifacts import ImageArtifact
+from griptape.drivers import BaseImageGenerationDriver
+
+
+@define
+class OpenAiDalleImageGenerationDriver(BaseImageGenerationDriver):
+    """Driver for OpenAI DALLE image generation API.
+
+    Attributes:
+        model: OpenAI DALLE model, for example 'dall-e-2' or 'dall-e-3'.
+        api_type: OpenAI API type, for example 'open_ai' or 'azure'.
+        api_version: API version.
+        base_url: API URL.
+        api_key: OpenAI API key.
+        organization: OpenAI organization ID.
+        style: Optional and only supported for dall-e-3, can be either 'vivid' or 'natural'.
+        quality: Optional and only supported for dall-e-3. Accepts 'standard', 'hd'.
+        image_size: Size of the generated image. Must be one of the following, depending on the requested model:
+            dall-e-2: [256x256, 512x512, 1024x1024]
+            dall-e-3: [1024x1024, 1024x1792, 1792x1024]
+        response_format: The response format. Currently only supports 'b64_json' which will return
+            a base64 encoded image in a JSON object.
+    """
+
+    api_type: str = field(default=openai.api_type, kw_only=True)
+    api_version: Optional[str] = field(default=openai.api_version, kw_only=True)
+    base_url: str = field(default=None, kw_only=True)
+    api_key: Optional[str] = field(default=None, kw_only=True)
+    organization: Optional[str] = field(default=openai.organization, kw_only=True)
+    client: openai.OpenAI = field(
+        default=Factory(
+            lambda self: openai.OpenAI(api_key=self.api_key, base_url=self.base_url, organization=self.organization),
+            takes_self=True,
+        )
+    )
+    style: Optional[str] = field(default=None, kw_only=True)
+    quality: Literal["standard"] | Literal["hd"] = field(default="standard", kw_only=True)
+    image_size: (
+        Literal["256x256"] | Literal["512x512"] | Literal["1024x1024"] | Literal["1024x1792"] | Literal["1792x1024"]
+    ) = field(default="1024x1024", kw_only=True)
+    response_format: Literal["b64_json"] = field(default="b64_json", kw_only=True)
+
+    def generate_image(self, prompts: list[str], **kwargs) -> ImageArtifact:
```

Comment:
Missing `negative_prompts`?

Reply:
OpenAI's API doesn't support negative prompts. The prompt is rewritten, so the specifics of any 'do not' instructions have an opportunity to be warped by rewriting or summarization. From my experimentation negative prompts in a unified prompt seem to be counterproductive, like `negative prompts: text, clouds` results in more text and clouds in the image than not including the negative prompts at all.

Reply:
I think the method signature should still include it though, right? Maybe we throw an error if it's provided.

Reply:
I realize now that you might just mean in the signature (i.e. instead of `**kwargs`). Updated to make that change. If I'm mistake, let me know.